### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by ANeaves**
Created the Integration testing for Mercury, using the Odin Data Framework. Also created an add-on Python script that does some basic checking of the resulting data.

### CODE REVIEW

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp [hunk 1]
```diff
@@ -88,16 +128,16 @@ namespace FrameSimulator {
     void MercuryFrameSimulatorPlugin::create_frames(const int &num_frames) {
         LOG4CXX_DEBUG(logger_, "Creating Frames");
 
-        // Build array of pixel data used for each frame
-        const int num_pixels = Mercury::pixel_columns_per_sensor * Mercury::pixel_rows_per_sensor;
-        uint16_t* pixel_data = new uint16_t[num_pixels];
+        // Build array of pixel data used for each frame NOW COVERED IN READING JSON
```
Comment (timcnicholls) [inline data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: SHOUTY COMMENT :wink:

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp [hunk 2]
```diff
@@ -88,16 +128,16 @@ namespace FrameSimulator {
...
+        // const int num_pixels_ = Mercury::pixel_columns_per_sensor * Mercury::pixel_rows_per_sensor;
```
Comment (timcnicholls) [inline data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: If commented out code is no longer needed it should be removed ideally

File: data/test/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+add_subdirectory(intergrationTest)
```
Comment (timcnicholls) [inline data/test/CMakeLists.txt]: Not a comment on this file specifically, but you have a typo in the path data/test/intergrationTest - extra 'r' in integration

**Review by timcnicholls — COMMENTED** [review]
Looks good 👍 - some minor things spotted

File: data/test/CMakeLists.txt (hunk 3 above)
Comment (ANeaves) [reply data/test/CMakeLists.txt]: WOOPS

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp (hunk 2 above)
Comment (ANeaves) [reply data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: Aye, thought I'd done that but it slipped through the net. Done

File: data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp (hunk 1 above)
Comment (ANeaves) [reply data/frameSimulator/src/MercuryFrameSimulatorPlugin.cpp]: ITS GONE NOW ALONG WITH THE REDUNDANT COMMENTED OUT CODE IT WAS SHOUTING ABOUT

**Review by timcnicholls — APPROVED** [review]

File: data/frameProcessor/src/CMakeLists.txt [hunk 4]
```diff
@@ -7,7 +7,7 @@ include_directories(${FRAMEPROCESSOR_DIR}/include ${ODINDATA_INCLUDE_DIRS}
 
 # Add library for each mercury plugin
 add_library(MercuryTemplatePlugin SHARED MercuryTemplatePlugin.cpp)
-add_library(MercuryProcessorPlugin SHARED MercuryProcessorPlugin.cpp)
+# add_library(MercuryProcessorPlugin SHARED MercuryProcessorPlugin.cpp)
```
Comment (dw96) [inline data/frameProcessor/src/CMakeLists.txt]: Can commented out line be removed if no longer needed?

File: data/frameProcessor/src/CMakeLists.txt [hunk 5]
```diff
@@ -17,7 +17,7 @@ add_library(MercuryNextFramePlugin SHARED MercuryNextFramePlugin.cpp)
 add_library(MercuryThresholdPlugin SHARED MercuryThresholdPlugin.cpp)
 
 install(TARGETS MercuryTemplatePlugin LIBRARY DESTINATION lib)
-install(TARGETS MercuryProcessorPlugin LIBRARY DESTINATION lib)
+# install(TARGETS MercuryProcessorPlugin LIBRARY DESTINATION lib)
```
Comment (dw96) [inline data/frameProcessor/src/CMakeLists.txt]: Similarly, no longer needed?

File: data/frameSimulator/include/MercuryFrameSimulatorPlugin.h [hunk 6]
```diff
@@ -19,20 +22,23 @@ using namespace log4cxx::helpers;
 
 namespace FrameSimulator {
 
-
-
     /** MercuryFrameSimulatorPlugin
      *
      *  'extract_frames' is called on setup: this takes the content of the pcap file and reproduces the
      *  excalibur frames to store
```
Comment (dw96) [inline data/frameSimulator/include/MercuryFrameSimulatorPlugin.h]: Reference to Excalibur, is this correct or should it read "Mercury" instead?

File: data/test/integrationTest/dataset_check.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        # read datasets into a variable of some sort
+        with h5py.File(args.filename, "r") as data_file:
+            print("Keys: {}".format(data_file.keys()))
+
+            self.data = np.array(data_file[args.dataset_name])
+            self.raw = np.array(data_file["raw_frames"])
+            # print(self.data)
+
+    def check_averages(self):
+
```
Comment (dw96) [inline data/test/integrationTest/dataset_check.py:23]: It all looks good but I'm just wondering whether the function should be named something else if you have any plans on implementing something similar for the Discrimination algorithm?

**Review by dw96 — APPROVED** [review]
It all looks really good, numpy really makes life a lot easier doesn't it?

File: data/test/integrationTest/dataset_check.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    def __init__(self, args):
+
+        # open file
+        # read datasets into a variable of some sort
+        with h5py.File(args.filename, "r") as data_file:
+            print("Keys: {}".format(data_file.keys()))
+
+            self.data = np.array(data_file[args.dataset_name])
+            self.raw = np.array(data_file["raw_frames"])
+            # print(self.data)
```
Comment (dw96) [inline data/test/integrationTest/dataset_check.py:20]: Just spotted this, can be removed if commented out?

**Review by dw96 — APPROVED** [review]
Just one more thing (in the voice of Colombo?)

**Pull request merged**